iris-mpc/bin/server.rs

```diff
@@ -693,6 +694,79 @@ fn get_iris_shares_parse_task(
     Ok(handle)
 }
 
+async fn send_last_modifications_to_sns(
+    store: &Store,
+    sns_client: &SNSClient,
+    config: &Config,
+    reauth_message_attributes: &HashMap<String, MessageAttributeValue>,
+    deletion_message_attributes: &HashMap<String, MessageAttributeValue>,
+    lookback: usize,
+) -> eyre::Result<()> {
+    // Fetch the last modifications from the database
+    let last_modifications = store.last_modifications(lookback).await?;
+
+    if last_modifications.is_empty() {
+        tracing::info!("No last modifications found to send to SNS");
+        return Ok(());
+    }
+
+    // Collect messages by type
+    let mut deletion_messages = Vec::new();
+    let mut reauth_messages = Vec::new();
+    for modification in &last_modifications {
+        let body = modification
+            .sns_message_body
+            .as_ref()
+            .expect("Missing SNS message body")
+            .clone();
+
+        match modification.request_type.as_str() {
+            IDENTITY_DELETION_MESSAGE_TYPE => {
+                deletion_messages.push(body);
+            }
+            REAUTH_MESSAGE_TYPE => {
+                reauth_messages.push(body);
+            }
+            other => {
+                tracing::error!("Unknown message type: {}", other);
+            }
+        }
+    }
+
+    tracing::info!(
+        "Sending {} last modifications to SNS. {} deletion, {} reauth",
+        last_modifications.len(),
+        deletion_messages.len(),
+        reauth_messages.len(),
+    );
+
+    if !deletion_messages.is_empty() {
+        send_results_to_sns(
+            deletion_messages,
+            &Vec::new(), // Not sure what these "empty" messages represent
```

Comment:
this is the request metadata (node_id, trace_id, span_id) that is added to the sns message in case it is present.
I guess since it is not serialized into the DB we cannot send the original here (also it might not even make sense to send the orginal trace information)